dashboard/src/app/(authenticated)/components/tables/WfSpecTable.tsx

```diff
@@ -1,30 +1,57 @@
+import { FC, Fragment, useMemo } from 'react'
+import { SearchResultProps } from '.'
 import { WfSpecId } from 'littlehorse-client/proto'
+
 import { TagIcon } from 'lucide-react'
-import Link from 'next/link'
-import { FC, Fragment } from 'react'
-import { SearchResultProps } from '.'
+import { Separator } from '@/components/ui/separator'
+import { useRouter } from 'next/navigation'
 
 export const WfSpecTable: FC<SearchResultProps> = ({ pages = [] }) => {
+  const router = useRouter()
+
   if (pages.length === 0) {
     return <div className="flex min-h-[360px] items-center justify-center text-center italic">No WfSpecs</div>
   }
 
+  const wfSpecs = useMemo(() => {
+    const specMap = new Map<string, wfSpecData>()
+
+    pages
+      .flatMap(page => page.results)
+      .reverse()
+      .forEach(({ name, majorVersion, revision }: WfSpecId) => {
+        if (!specMap.has(name)) {
+          specMap.set(name, { name, latestVersion: `${majorVersion}.${revision}` })
+        }
+      })
+
+    return Array.from(specMap.values())
+  }, [pages])
+
```

Comment:
I see there might be cases where the results might not be complete due to pagination. So I suspect it would get the latest version wrong. 

Have you tried that?

Reply:
> I see there might be cases where the results might not be complete due to pagination. So I suspect it would get the latest version wrong.
> 
> Have you tried that?

This is a valid issue that you bring up. After analyzing the data that is sent to the front end page, I don't believe there is a suitable solution for us here. It seems like a limitation from the server and the data being sent. 

How do you think we could solve this?

Reply:
This is also an issue the server has, @coltmcnealy-lh and I talked about this in terms of of trying you query user tasks for "the latest 10" and the problem is the server can't do that. I'm not sure if this issue is related but I believe this issue might be easier to solve as the server could have a different RPC that just sends WFSpec and its latest version instead of each individual WfSpec

Reply:
To clarify the situation here for anyone else reading this:

![image](https://github.com/user-attachments/assets/e75cc0b5-4097-4589-9cb1-3d86cadfaded)

There's a pagination problem with WfSpecs display. The page shows WfSpecs with a limit on how many items load at once (`items per load`), but the UI incorrectly assumes that the most recent version of a WfSpec will always be within this loaded set.

Example of the problem:
If items_per_load = 10
- The latest version might actually be the 11th item or later
- Current UI would miss this latest version by only looking at the first 10 loaded items

This is a design flaw because the UI needs to account for cases where the latest version exists beyond the initial pagination limit.

Reply:
There is `rpc GetLatestWfSpec` that gets the latest version of a `WfSpec`

Reply:
@bryson-g and @HazimAr  check out this RPC: https://littlehorse.dev/docs/api#getlatestwfspec

Reply:
We can also add a flag to `rpc SearchWfSpec` that filters so that it only returns the latest version of each `WfSpec`. That might be the easiest solution, but it requires backend changes. @HazimAr maybe that would be a good "warmup" for you to get used to the internals of the server.

Alternatively, this would probably be a half-day of effort for @Snarr

Reply:
I agree it would be a good warmup but I don't think that is necessary now

Also if we used GetLatestWfSpec then it will fix the problem of being uncertain of the latest version but the per load will still be off 

Because it would return 10 items but 8 are unique so it would call 8 times to get the latest version